src/core/qgssnappingconfig.h

```diff
@@ -70,11 +70,12 @@ class CORE_EXPORT QgsSnappingConfig
     enum SnappingTypes
     {
       NoSnapFlag = 0, //!< No snapping
-      VertexFlag = 1, //!< On vertices
-      SegmentFlag = 2, //!< On segments
-      AreaFlag = 4, //!< On Area
-      CentroidFlag = 8, //!< On centroid
-      MiddleOfSegmentFlag = 16, //!< On Middle segment
+      VertexFlag = 1 << 0, //!< On vertices
+      SegmentFlag = 1 << 1, //!< On segments
+      AreaFlag = 1 << 2, //!< On Area
+      CentroidFlag = 1 << 3, //!< On centroid
+      MiddleOfSegmentFlag = 1 << 4, //!< On Middle segment
+      LineEndpointFlag = 1 << 5, //!< Start or end points of lines only (since QGIS 3.20)
```

Comment:
Maybe also document its use on polygon layers? Or skip the *of lines only*?

Reply:
Done